### PR TITLE
It keeps say 'Cannot find git-hooks. Did you install it?' #6

### PR DESCRIPTION
**Pull request by JamesMcGuigan**
One of the common getting started errors is non-executable permissions on the hook script files. I have added a user friendly try/catch error message to hint at this possible problem. I have not tested this on windows, and I have not done an exhaustive list of all the reasons child_process.spawn() might throw an exception, assuming a permissions issue is the primary cause.

This however is a much less obscure error message than:
```Cannot find git-hooks. Did you install it?```

https://github.com/tarmolov/git-hooks-js/issues/6

### CODE REVIEW

**Comment by JamesMcGuigan** [comment]
Example error message is:
```git-hooks unable to execute file: /Users/jamie/Edgefolio/source/webapps/.githooks/pre-commit/npm-shrinkwrap.sh with permissions: 644 is the file executable?```

**Comment by tarmolov** [comment]
I cannot approve the pr because of the following issues:
  * `.idea` in `.gitignore` is unnecessary. You have to ignore it in your user `.gitignore`.
  * `git-hooks` shouldn't catch errors because it is `hook-template.js` job.
  * `git-hooks` doesn't use sync methods.

I took advantage of your pr and fixed the issue. Thank you!

**Comment by tarmolov** [comment]
1.0.1

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 !/node_modules/git-hooks
 /coverage
```
Comment (tarmolov) [inline .gitignore]: test

File: lib/git-hooks.js [hunk 2]
```diff
@@ -125,14 +125,22 @@ function runHooks(hooks, args, callback) {
         return;
     }
 
-    var hook = spawn(hooks.shift(), args, {stdio: 'inherit'});
```
Comment (tarmolov) [inline lib/git-hooks.js]: test

Comment (tarmolov) [inline lib/git-hooks.js]: test

File: lib/git-hooks.js [hunk 3]
```diff
@@ -125,14 +125,22 @@ function runHooks(hooks, args, callback) {
...
-    hook.on('close', function (code) {
-        if (code === 0) {
-            runHooks(hooks, args, callback);
-        } else {
-            callback(code);
-        }
-    });
+    var scriptFile = hooks.shift();
```
Comment (tarmolov) [inline lib/git-hooks.js]: test

File: lib/git-hooks.js [hunk 4]
```diff
@@ -125,14 +125,22 @@ function runHooks(hooks, args, callback) {
...
     }
 
-    var hook = spawn(hooks.shift(), args, {stdio: 'inherit'});
-    hook.on('close', function (code) {
-        if (code === 0) {
-            runHooks(hooks, args, callback);
-        } else {
-            callback(code);
-        }
-    });
```
Comment (tarmolov) [inline lib/git-hooks.js]: test

File: lib/git-hooks.js [hunk 5]
```diff
@@ -125,14 +125,22 @@ function runHooks(hooks, args, callback) {
...
+    var scriptFile = hooks.shift();
+    try {
+        var hook = spawn(scriptFile, args, {stdio: 'inherit'});
+        hook.on('close', function (code) {
+            if (code === 0) {
+                runHooks(hooks, args, callback);
+            } else {
+                callback(code);
```
Comment (tarmolov) [inline lib/git-hooks.js]: test

**Pull request closed without merging**